File: Add_Two_Numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode ans = new ListNode((l1.val + l2.val) %10 );
        ListNode current = ans;
        ListNode run1 = l1.next;
        ListNode run2 = l2.next;
        int carry = (l1.val + l2.val) / 10;
        while (run1 != null && run2 != null) {
            ListNode add = new ListNode((run1.val + run2.val + carry) % 10 );
            carry = (run1.val + run2.val + carry)/10;
            run1 = run1.next;
            run2 = run2.next;
            current.next = add;
            current = current.next;
        }
        if (run1 == null) {
            ListNode temp = run1;
            run1 = run2;
            run2 = temp;
        }
        while (run1 != null) {
            ListNode add = new ListNode((run1.val + carry) % 10 );
            carry = (run1.val + carry)/10;
            run1 = run1.next;
            current.next = add;
            current = current.next;
        }
        if (carry != 0) {
            ListNode last = new ListNode(carry );
            current.next = last;
            current = current.next;
        }


        return ans;
    }
}

File: Add_Two_Numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        ans = ListNode((l1.val + l2.val) %10 )
        current = ans
        run1 = l1.next
        run2 = l2.next
        carry = (l1.val + l2.val) / 10
        while (run1 is not None and run2 is not None):
            add = ListNode((run1.val + run2.val + carry) % 10 )
            carry = (run1.val + run2.val + carry)/10
            run1 = run1.next
            run2 = run2.next
            current.next = add
            current = current.next

        if (run1 is None) :
            temp = run1
            run1 = run2
            run2 = temp

        while (run1 is not None) :
            add = ListNode((run1.val + carry) % 10 )
            carry = (run1.val + carry)/10
            run1 = run1.next
            current.next = add
            current = current.next

        if (carry is not 0) :
            last = ListNode(carry )
            current.next = last
            current = current.next





        return ans
